src/physics/em/detail/UrbanMscScatter.hh

```diff
@@ -76,13 +61,19 @@ class UrbanMscScatter
     // Urban MSC helper class
     UrbanMscHelper helper_;
     // Results from UrbanMSCStepLimit
-    StepLimitResult input_;
+    MscStep input_;
     // Geomtry track view
     GeoTrackView& geometry_;
 
+    real_type lambda_;
+    real_type true_path_;
+
+    // Internal state
+    real_type tau_{0};
+
     //// COMMON PROPERTIES ////
 
-    //! The minimum step length for geometry ( 0.05*CLHEP::nm)
+    //! The minimum step length for geometry 0.05 nm
```

Comment:
I find that this should be same as UrbanMscParameters::geom_limit (which I mistakenly introduced it redundantly).  Can you choose one of them and remove the other?